packages/admin-next/dashboard/src/routes/orders/order-create-fulfillment/components/order-create-fulfillment-form/order-create-fulfillment-form.tsx

```diff
@@ -0,0 +1,192 @@
+  })
+
+  return (
+    <RouteFocusModal.Form form={form}>
+      <form
+        onSubmit={handleSubmit}
+        className="flex h-full flex-col overflow-hidden"
+      >
+        <RouteFocusModal.Header>
+          <div className="flex items-center justify-end gap-x-2">
+            <RouteFocusModal.Close asChild>
+              <Button size="small" variant="secondary">
+                {t("actions.cancel")}
+              </Button>
+            </RouteFocusModal.Close>
+            <Button size="small" type="submit" isLoading={isMutating}>
+              {t("orders.fulfillment.create")}
+            </Button>
+          </div>
+        </RouteFocusModal.Header>
+        <RouteFocusModal.Body className="flex h-full w-full flex-col items-center divide-y overflow-y-auto">
+          <div className="flex size-full flex-col items-center overflow-auto p-16">
+            <div className="flex w-full max-w-[736px] flex-col justify-center px-2 pb-2">
+              <div className="flex flex-col divide-y">
+                <div className="flex-1">
+                  <Form.Field
+                    control={form.control}
+                    name="location_id"
+                    render={({ field: { onChange, ref, ...field } }) => {
+                      return (
+                        <Form.Item>
+                          <Form.Label> {t("fields.location")}</Form.Label>
+                          <Form.Hint>
+                            {t("orders.fulfillment.locationDescription")}
+                          </Form.Hint>
+                          <Form.Control>
+                            <Select onValueChange={onChange} {...field}>
+                              <Select.Trigger
+                                className="bg-ui-bg-base"
+                                ref={ref}
+                              >
+                                <Select.Value />
+                              </Select.Trigger>
+                              <Select.Content>
+                                {stock_locations.map((l) => (
+                                  <Select.Item key={l.id} value={l.id}>
+                                    {l.name}
+                                  </Select.Item>
+                                ))}
+                              </Select.Content>
+                            </Select>
+                          </Form.Control>
+                          <Form.ErrorMessage />
+                        </Form.Item>
+                      )
+                    }}
+                  />
+
+                  <Form.Item className="mt-8">
+                    <Form.Label>
+                      {t("orders.fulfillment.itemsToFulfill")}
+                    </Form.Label>
+                    <Form.Hint>
+                      {t("orders.fulfillment.itemsToFulfillDesc")}
+                    </Form.Hint>
+
+                    <div className="flex flex-col gap-y-1">
+                      {fulfillableItems.map((item) => (
```

Comment:
suggestion: there is no way to go back after deleting all the items, which puts the form in a non-submittable state. I think it might make sense to add a "Reset" button or something along those lines